src/base/contracts/baseContract.ts

```diff
@@ -1,47 +1,82 @@
-import * as fs from 'fs';
 import * as path from 'path';
-import { ensure } from '../util/ensure';
-import { isValidJSON } from '../util/json';
-import { ContractABI, MethodABI } from './json';
-import { MethodContainerMap, MethodManager } from './method/manager';
+import * as fs from 'fs';
+import { ensure } from '../../util/ensure';
+import { isValidJSON } from '../../util/json';
+import { BaseMethodManager, BaseMethodContainerMap } from '../method/baseManager';
+import { ContractABI, MethodABI } from '../iface';
 
 /**
  * A high-level wrapper for Solidity smart contract function calls
  */
-export class Contract {
-  static readonly ABI_DIR = '../../abis';
-  static readonly CONFIG_DIR = '../../config';
-  static readonly DEFAULT_INSTANCE_KEY = 'default';
+export abstract class BaseContract< K extends MethodABI, T extends ContractABI> {
```

Comment:
nit: spacing

Reply:
Thank you, with this comment we realize others lint errors, we run npm run lint to catch them